### PR TITLE
fix test_client connected error

### PR DESCRIPTION
**Pull request by shinoi2**
The server disconnects all clients, only one testclient will disconnect.

### CODE REVIEW

**Comment by miguelgrinberg** [comment]
I don't understand what is the issue that this change addresses. Can you provide an example please?

**Comment by shinoi2** [comment]
I add an example in test.

File: test_socketio.py [hunk 1]
```diff
@@ -237,7 +237,7 @@ def on_json(self, data):
             return data
 
     def on_exit(self, data):
-        disconnect()
```
Comment (miguelgrinberg) [inline test_socketio.py:240]: this change seems unrelated

File: src/flask_socketio/test_client.py [hunk 2]
```diff
@@ -203,8 +204,6 @@ def get_received(self, namespace=None):
                           namespace is assumed if this argument is not
                           provided.
         """
-        if not self.is_connected(namespace):
-            raise RuntimeError('not connected')
```
Comment (miguelgrinberg) [inline src/flask_socketio/test_client.py:207]: Why was this removed?

**Comment by miguelgrinberg** [comment]
Thanks, I do understand the problem now. I have addressed in a slightly different way.
